src/app/services/admin-patient.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { ServerApiService } from './server-api.service';
+import { Observable } from 'rxjs';
+
+export class GenericAdminPatientResponse {
+  public message: string;
+}
+
+export class ResetPasswordResponse {
+  public password: string;
+}
+
+export class UserListItem {
+  public name: string;
+  public type: string;
+}
+
+@Injectable()
+export class AdminPatientService {
+  constructor(private serverApiService: ServerApiService) { }
+
+  public createUser(name: string, password: string, type: string) {
+    return this.serverApiService.post<GenericAdminPatientResponse>('/user/create', {'name': name, 'password': password, 'type': type});
+  }
+
+  public renameUser(currName: string, nextName: string) {
+    return this.serverApiService.post<GenericAdminPatientResponse>('/user/rename', {'oldName': currName, 'newName': nextName});
+  }
+
+  public resetPassword(name: string) {
+    return this.serverApiService.post<ResetPasswordResponse>('/user/resetPassword', { 'name': name });
+  }
+
+  public deleteUser(name: string) {
+    let params : Map<string, string> = new Map();
```

Comment:
you can simplify this map constructor
```suggestion
    let params : Map<string, string> = new Map([['name', name]);
```